lib/vfscore/vfs.h

```diff
@@ -69,89 +84,819 @@ extern int vfs_debug;
+ *			if offset + len is greater than the file size.
+ *		* FALLOC_FL_PUNCH_HOLE - deallocates len bytes starting at
+ *			offset and creates a hole in the file. This flag should
+ *			always come together with FALLOC_FL_KEEP_SIZE, otherwise
+ *			the sys_futimens call will fail.
+ * @param offset
+ *	Starting byte from which this operation would apply
+ * @param len
+ *	Number of bytes for which this operation would apply
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int sys_fallocate(struct vfscore_file *fp, int mode, loff_t offset, loff_t len);
 
-int	 task_alloc(struct task **pt);
-int	 task_conv(struct task *t, const char *path, int mode, char *full);
-int	 path_conv(char *wd, const char *cpath, char *full);
+/**
+ * This function is not used at this point.
+ */
+int sys_pivot_root(const char *new_root, const char *old_put);
 
-//int	 sec_file_permission(task_t task, char *path, int mode);
-int	 sec_vnode_permission(char *path);
+/**
+ * Calls the wrapper Unikraft sync() syscall function.
+ *
+ */
+void sync(void);
 
-int     namei(const char *path, struct dentry **dpp);
-int	 namei_last_nofollow(char *path, struct dentry *ddp, struct dentry **dp);
-int	 lookup(char *path, struct dentry **dpp, char **name);
-void	 vnode_init(void);
-void	 lookup_init(void);
+/**
+ * Changes mode of the file whose pathname is given in pathname.
+ *
+ * @param path
+ *	Path to the file
+ * @param mode
+ *	The new file mode
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int	sys_chmod(const char *path, mode_t mode);
 
-int     vfs_findroot(const char *path, struct mount **mp, char **root);
-int	 vfs_dname_copy(char *dest, const char *src, size_t size);
+/**
+ * Changes mode of the file specified by the open file descriptor fd.
+ *
+ * @param fd
+ *	An open file descriptor
+ * @param mode
+ *	The new file mode is specified in mode
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int sys_fchmod(int fd, mode_t mode);
 
-int	 fs_noop(void);
+/**
+ *  * Converts to full path from the cwd of task and path.
+ *
+ * @param t
+ *	Pointer to task containing the cwd
+ * @param path
+ *	The target path
+ * @param mode
+ *	Unused parameter
+ * @param[out] full
+ *	Full path to be returned
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int task_conv(struct task *t, const char *path, int mode, char *full);
+
+/**
+ * Converts to full path from the wd of task and cpath.
+ *
+ * @param wd
+ *	Working directory
+ * @param cpath
+ *	The target path
+ * @param[out] full
+ *	Full path to be returned
+ * @return
+ *	- (0):  Successfully initialized
+ *	- (<0): Negative value with error code
+ */
+int path_conv(char *wd, const char *cpath, char *full);
+
+/**
+ * This function is not implemented or used at this point.
+ */
+int sec_vnode_permission(char *path);
```

Comment:
Do we need to keep the definition of this file here? If it is not used or implemented should we just delete it?
(cc. @razvand)

Reply:
Yes, just leave it. Maybe we can do a deeper investigation at some other time.